python/foglamp/services/core/api/notification.py

```diff
@@ -0,0 +1,517 @@
+        return web.json_response({'result': "No Notification service available."})
+
+    try:
+        url = 'http://{}:{}/foglamp/notification/rules'.format(_address, _port)
+        rule_plugins = json.loads(await _hit_get_url(url))
+
+        url = 'http://{}:{}/foglamp/notification/delivery'.format(_address, _port)
+        delivery_plugins = json.loads(await _hit_get_url(url))
+    except Exception as ex:
+        raise web.HTTPInternalServerError(reason=ex)
+    else:
+        return web.json_response({'rules': rule_plugins, 'delivery': delivery_plugins})
+
+
+async def get_notification(request):
+    """ GET an existing notification
+
+    :Example:
+        curl -X GET http://localhost:8081/foglamp/notification/<notification_name>
+    """
+    try:
+        notif = request.match_info.get('notification_name', None)
+        if notif is None:
+            raise web.HTTPInternalServerError(reason="Notification name is required.")
+
+        notification = {}
+        storage = connect.get_storage_async()
+        config_mgr = ConfigurationManager(storage)
+        notification_config = await config_mgr._read_category_val(notif)
+        if notification_config:
+            rule_config = await config_mgr._read_category_val("rule{}".format(notif))
+            channel_config = await config_mgr._read_category_val("delivery{}".format(notif))
+            notification = {
+                "name": notification_config['name']['value'],
+                "description": notification_config['description']['value'],
+                "rule": notification_config['rule']['value'],
+                "ruleConfig": rule_config,
+                "channel": notification_config['channel']['value'],
+                "deliveryConfig": channel_config,
+                "notification_type": notification_config['notification_type']['value'],
+                "enable": notification_config['enable']['value'],
+            }
+    except Exception as ex:
+        raise web.HTTPInternalServerError(reason=ex)
+    else:
+        return web.json_response({'notification': notification})
+
+
+async def get_notifications(request):
+    """ GET list of notifications
+
+    :Example:
+        curl -X GET http://localhost:8081/foglamp/notification
+    """
+    try:
+        storage = connect.get_storage_async()
+        config_mgr = ConfigurationManager(storage)
+        all_notifications = await config_mgr._read_all_child_category_names("Notifications")
+        notifications = []
+        for notification in all_notifications:
+            notification_config = await config_mgr._read_category_val(notification['child'])
+            notification = {
+                "name": notification_config['name']['value'],
+                "rule": notification_config['rule']['value'],
+                "channel": notification_config['channel']['value'],
+                "notification_type": notification_config['notification_type']['value'],
```

Comment:
We followed `camelCase` convention when returning API response. (i.e notificationType)

Reply:
done